internal/cache/in_memory.go

```diff
@@ -0,0 +1,50 @@
+package cache
+
+import (
+	"sync"
+	"time"
+)
+
+type MemoryStore struct {
+	data map[string][]byte
+	mu   sync.RWMutex
```

Comment:
As an alternative a `sync.Map` could be used. We already use that e.g. here: https://github.com/bitcoin-sv/arc/blob/6669b2339f9b019dec239acc264bf0214d705e33/internal/metamorph/response_processor.go#L44
or here: https://github.com/bitcoin-sv/arc/blob/4e031edcef23927fa0b79b82fbe50642dff0efbe/internal/broadcaster/rate_broadcaster.go#L42

The advantage of the sync map is that you don't need to use a separate mutex as it is concurrency safe. Here is more information on it: https://cs.opensource.google/go/go/+/go1.23.3:src/sync/map.go;l=38

Reply:
Done